### PR TITLE
Replace numerical status codes with http constants

### LINKED ISSUE #34154: Use HTTP status code constants instead of numeric values
Grafana backend code uses a mix of numeric status code (i.e. 200, 404, 500 etc) and http status code constants. Even though some might consider it a matter of personal preference, I suggest agreeing on a common coding style here.

It looks like newer source code tends to use http codes. Those are more descriptive, provide correct semantics and correct context for use, therefore - I would suggest replacing magic status code numbers with http package constants.

Here's some codebase archeology to estimate the scope of changes.

Status codes are mostly handled by the `pkg/api/response` package in `response.go` file - https://github.com/grafana/grafana/blob/main/pkg/api/response/response.go

Here we can see `status` parameter to be used in helper functions, such as `Error()`, `JSON()`, `JSONStreaming`, `Empty`, and `Respond`. There is also `CreateNormalResponse`, which does not use numerical status codes.

Numerical status code distribution is trivial to get with something like `grep 'response.\(Error\|JSON\|Respond\|JSONStreaming\|Empty\)(\d\d\d' -h -o -R . | sort | uniq -c`:

```
   1 response.Empty(200
   1 response.Error(200
   1 response.Error(501
   1 response.Error(503
   1 response.JSONStreaming(200
   1 response.JSONStreaming(404
   2 response.JSON(202
   2 response.JSON(404
   2 response.JSON(409
   3 response.JSON(503
   3 response.Respond(200
   4 response.JSON(412
   5 response.Error(422
   7 response.Error(412
   8 response.Error(409
  10 response.Error(401
  26 response.Error(403
  67 response.Error(404
  83 response.Error(400
 119 response.JSON(200
 234 response.Error(500
```

A few lines of `gofmt -w -r ...` should do the job.

### PR DESCRIPTION
**Pull request by zserge**
Closes #34154 

Automatic `gofmt` replacement of status code numbers with http package constants.

### CODE REVIEW

File: pkg/api/password.go [hunk 1]
```diff
@@ -13,22 +14,22 @@ import (
...
+		return response.Error(http.StatusUnauthorized, "Not allowed to reset password when login form is disabled", nil)
 	}
 
 	userQuery := models.GetUserByLoginQuery{LoginOrEmail: form.UserOrEmail}
 
 	if err := bus.Dispatch(&userQuery); err != nil {
 		c.Logger.Info("Requested password reset for user that was not found", "user", userQuery.LoginOrEmail)
-		return response.Error(200, "Email sent", err)
+		return response.Error(http.StatusOK, "Email sent", err)
 	}
```
Comment (zserge) [inline pkg/api/password.go:28]: Something fishy is happening here: returning Error, with status OK, "Email sent" message and "not found" log. I would like someone to confirm that this is correct.

**Review by aknuds1 — APPROVED** [review]
Great idea, but this stylistic choice should be coordinated with the rest of the team in e.g. a daily sync.

File: pkg/api/password.go (hunk 1 above)
Comment (aknuds1) [reply pkg/api/password.go:28]: @marefr should we use a non-error response (e.g. not `response.Error`) here? I assume it's for security reasons the response obscures what actually happens.

Comment (marefr) [reply pkg/api/password.go:28]: Looks a bit weird but think the reason is that the provided error will be automatically logged. You're correct in we don't want to return the actual problem to user due to security

**Comment by zserge** [comment]
Makes sense, then I will just leave a comment that this code snippet works
as intended.

**Comment by ying-jeanne** [comment]
Really nice refectory <3, it is probably not important, thinking should we regex also to make the unittest assert.Equal(t, 404, sc.resp.Code) -> assert.Equal(t, http.StatusNotFound, sc.resp.Code), just to make everything the same way.

**Comment by marefr** [comment]
What's the status of this one? Has it been coordinated/discussed with rest of the team?

**Review by ying-jeanne — APPROVED** [review]
LGTM

**Pull request closed without merging**